src/Internal/Service/PaymentProcessingService.php

```diff
@@ -24,13 +27,25 @@ class PaymentProcessingService {
 	 */
 	private $state_factory;
 
+	/**
+	 * Legacy Proxy.
+	 *
+	 * @var LegacyProxy
+	 */
+	private $legacy_proxy;
+
 	/**
 	 * Service constructor.
 	 *
 	 * @param StateFactory $state_factory Factory for payment states.
+	 * @param LegacyProxy  $legacy_proxy  Legacy proxy.
 	 */
-	public function __construct( StateFactory $state_factory ) {
+	public function __construct(
+		StateFactory $state_factory,
```

Comment:
Will every State have the same dependencies as these? Can we expose those into payment context and call them within the state if needed via payment context call?

Reply:
> Will every State have the same dependencies as these? 

No, states will be like controllers, and they will depend on varying services.

> Can we expose those into payment context and call them within the state if needed via payment context call?

We don't want the payment context to have any dependencies. Also, that would force all dependencies through the `PaymentContext`, and at some point, adding/modifying a dependency, would require fixing hundreds of tests. This is one of the issues with the current gateway classes, and one we definitely need to fix. I don't want to have to spend 2 days just fixing tests, only because I dared to add a simple dependency.

Reply:
@zmaglica should I consider this one resolved?